hector_gazebo_plugins/src/gazebo_ros_sonar.cpp

```diff
@@ -84,8 +86,22 @@ void GazeboRosSonar::Load(sensors::SensorPtr _sensor, sdf::ElementPtr _sdf)
   frame_id_ = "/sonar_link";
 
   // load parameters
-  if (_sdf->HasElement("robotNamespace"))
-    namespace_ = _sdf->GetElement("robotNamespace")->GetValue()->GetAsString();
+ if (_sdf->HasElement("useNamespaceFromScopedName")){
+     if (_sdf->GetElement("useNamespaceFromScopedName")){
```

Comment:
This should be an and `&&`. Otherwise, the option existing but set to false would not initialize the namespace at all.
Also, I'm pretty sure you also have to extract the value using `GetValue()->GetAsBool()`.

And please follow the code style of the project, e.g., brackets `{}` on new lines, no spaces between function names and parameters etc.